projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -505,4 +505,50 @@ public function add_gutenberg_menus( $wp_admin = false ) {
 	public function should_link_to_wp_admin() {
 		return get_user_option( 'jetpack_admin_menu_link_destination' );
 	}
+
+	/**
+	 * Returns the current slug from the URL.
+	 *
+	 * @return string
+	 */
+	public function get_current_slug() {
+		global $pagenow;
+		$slug = $pagenow;
+		if ( isset( $_REQUEST['post_type'] ) ) {
+			$slug .= '?post_type=' . $_REQUEST['post_type'];
+		}
+		if ( isset( $_REQUEST['taxonomy'] ) ) {
+			$slug .= '?taxonomy=' . $_REQUEST['taxonomy'];
+		}
+		if ( isset( $_REQUEST['page'] ) ) {
+			$slug .= '?page=' . $_REQUEST['page'];
+		}
+
+		return $slug;
+	}
```

Comment:
@mmtr we also needed this logic here, so we moved it to a function which hopefully you can reuse in https://github.com/Automattic/jetpack/pull/20147